src/bo4e/bo/regionaltarif.py

```diff
@@ -0,0 +1,86 @@
+"""
+
+from datetime import datetime
+from typing import List, Optional
+
+import attr
+from marshmallow import fields
+
+from bo4e.bo.tarifinfo import Tarifinfo, TarifinfoSchema
+from bo4e.com.regionalepreisgarantie import RegionalePreisgarantie, RegionalePreisgarantieSchema
+from bo4e.com.regionaleraufabschlag import RegionalerAufAbschlag, RegionalerAufAbschlagSchema
+from bo4e.com.regionaletarifpreisposition import RegionaleTarifpreisposition, RegionaleTarifpreispositionSchema
+from bo4e.com.tarifberechnungsparameter import Tarifberechnungsparameter, TarifberechnungsparameterSchema
+from bo4e.com.tarifeinschraenkung import Tarifeinschraenkung, TarifeinschraenkungSchema
+from bo4e.enum.botyp import BoTyp
+from bo4e.validators import check_list_length_at_least_one
+
+
+# pylint: disable=too-few-public-methods, empty-docstring
+@attr.s(auto_attribs=True, kw_only=True)
+class Regionaltarif(Tarifinfo):
+    # no description in the official docs.
+    # https://github.com/Hochfrequenz/BO4E-python/issues/338
+    """ """
+
+    bo_typ: BoTyp = attr.ib(default=BoTyp.REGIONALTARIF)
+    # required attributes
+    #: Gibt an, wann der Preis zuletzt angepasst wurde
+    preisstand: datetime = attr.ib(validator=attr.validators.instance_of(datetime))
+    #: Für die Berechnung der Kosten sind die hier abgebildeten Parameter heranzuziehen
+    berechnungsparameter: Tarifberechnungsparameter = attr.ib(
+        validator=attr.validators.instance_of(Tarifberechnungsparameter)
+    )
+    #: Die festgelegten Preise mit regionaler Eingrenzung, z.B. für Arbeitspreis, Grundpreis etc.
+    tarifpreise: List[RegionaleTarifpreisposition] = attr.ib(
+        validator=attr.validators.deep_iterable(
+            member_validator=attr.validators.instance_of(RegionaleTarifpreisposition),
+            iterable_validator=check_list_length_at_least_one,
+        )
+    )
+
+    # optional attributes
+    #: Auf- und Abschläge auf die Preise oder Kosten mit regionaler Eingrenzung
+    tarif_auf_abschlaege: Optional[List[RegionalerAufAbschlag]] = attr.ib(
+        default=None,
+        validator=attr.validators.optional(
+            attr.validators.deep_iterable(
+                member_validator=attr.validators.instance_of(RegionalerAufAbschlag),
+                iterable_validator=attr.validators.instance_of(list),
+            )
+        ),
+    )
+    #: Festlegung von Garantien für bestimmte Preisanteile
+    preisgarantien: Optional[List[RegionalePreisgarantie]] = attr.ib(
+        default=None,
+        validator=attr.validators.optional(
+            attr.validators.deep_iterable(
+                member_validator=attr.validators.instance_of(RegionalePreisgarantie),
+                iterable_validator=attr.validators.instance_of(list),
+            )
+        ),
+    )
+    #: Die Bedingungen und Einschränkungen unter denen ein Tarif angewendet werden kann
+    tarifeinschraenkungen: Optional[Tarifeinschraenkung] = attr.ib(
```

Comment:
hättre ich als singular benannt. würde ich als fehler in der doku deuten

Reply:
e136c58690a55d4498e8797615eaa7e24267b053